### DISCUSSION TITLE
TailwindCSS custom classes not working on Vercel

### DISCUSSION THREAD

**Discussion by fdarkaou** [tailwindlabs/tailwindcss#11479] [Help]
Hi everyone,

My custom Tailwindclasses and gradients are not being shown when deploying on Vercel, while they work perfectly on localhost (both dev and production). I am using Nuxt 2 with the latest TailwindCSS (3.3.2) being installed as a plugin.

This is how my nuxt.config.js looks like:

`css: ['~/assets/css/tailwind.css']`

  `build: {
    postcss: {
      plugins: [
        'tailwindcss',
        'postcss-flexbugs-fixes',
        [
          'postcss-preset-env',
          {
            autoprefixer: {
              flexbox: 'no-2009',
            },
            stage: 3,
            features: {
              'custom-properties': false,
            },
          },
        ],
      ],
    }
  }`

  tailwind.config.js:
`/** @type {import('tailwindcss').Config} */`

`module.exports = {
    content: [
      './components/**/*.{vue,js}',
      './layouts/**/*.vue',
      './pages/**/*.vue',
      './plugins/**/*.{js,ts}',
      './nuxt.config.{js,ts}',
      './assets/**/*.{vue,js}',
      './assets/css/*.css'
    ],
  safelist: ['text-gradient-warm', 'btn-gradient', 'btn-gradient:hover', 'bg-gradient-to-r', 'from-[#44BCFF]', 'via-[#FF44EC]', 'to-[#FF675E]'],
  theme: {
    extend: {
      colors: {
        'blue-custom': '#44BCFF',
        'pink-custom': '#FF44EC',
        'orange-custom': '#FF675E'
      }
    }
  },
  plugins: [
    require('daisyui'),
  ],
  daisyui: {
    styled: true,
    themes: ["light"],
  },
}
`

I have tried so many things but it just does not seem to work. I have also tried all solutions for similar problems I have found online but does not seem to be doing anything. Most text-gradients do not work, and most importantly, some gradient backgrounds on button do not work. For example:

`<div class="absolute transition-all duration-1000 opacity-60 -inset-px bg-gradient-to-r from-blue-custom via-pink-custom to-orange-custom rounded-xl blur-lg group-hover:opacity-80 group-hover:-inset-1 group-hover:duration-200 animate-tilt"></div>
`

Shows a nice gradient shadow underneath my button but it shows nothing once deployed on Vercel.

Would really appreciate any help on this as I have been stuck on it for days!

---

UPDATE: 
I just made a public repo ppl can clone: https://github.com/fdarkaou/tailwindvercel

You can all install dependencies using `yarn install`, and I am using `nuxt generate` as the build command.

You can see the difference between localhost and deployment on vercel here:

Vercel: https://pasteboard.co/ZBnG3n5BhWO8.png
Localhost: https://pasteboard.co/573VNtLDE9rQ.png

As you can see, the gradient is not showing up on Vercel.

Thank you!

**Comment by adamwathan** [comment]
Hey! If you can provide a complete reproduction (a project people can clone and try out themselves) it'll dramatically increase the odds of someone being able to help 👍

**Reply by fdarkaou** [reply]
Hi Adam, thanks for the tip. 

I just made a public repo ppl can clone: https://github.com/fdarkaou/tailwindvercel

You can all install dependencies using `yarn install`, and I am using `nuxt generate` as the build command.

You can see the difference between localhost and deployment on vercel here:

Vercel: https://pasteboard.co/ZBnG3n5BhWO8.png
Localhost: https://pasteboard.co/573VNtLDE9rQ.png

As you can see, the gradient is not showing up on Vercel.

Thank you!

**Comment by wongjn** [comment]
It seems like when deploying, some CSS variables from Tailwind are being stripped:

```diff

 *,
 ::before,
 ::after {
   --tw-border-spacing-x: 0;
   --tw-border-spacing-y: 0;
   --tw-translate-x: 0;
   --tw-translate-y: 0;
   --tw-rotate: 0;
   --tw-skew-x: 0;
   --tw-skew-y: 0;
   --tw-scale-x: 1;
   --tw-scale-y: 1;
-  --tw-pan-x: ;
-  --tw-pan-y: ;
   --tw-pinch-zoom: ;
   --tw-scroll-snap-strictness: proximity;
-  --tw-gradient-from-position: ;
-  --tw-gradient-via-position: ;
-  --tw-gradient-to-position: ;
-  --tw-ordinal: ;
-  --tw-slashed-zero: ;
-  --tw-numeric-figure: ;
-  --tw-numeric-spacing: ;
-  --tw-numeric-fraction: ;
-  --tw-ring-inset: ;
   --tw-ring-offset-width: 0px;
   --tw-ring-offset-color: #fff;
   --tw-ring-color: rgb(59 130 246 / 0.5);
   --tw-ring-offset-shadow: 0 0 #0000;
   --tw-ring-shadow: 0 0 #0000;
   --tw-shadow: 0 0 #0000;
   --tw-shadow-colored: 0 0 #0000;
-  --tw-blur: ;
-  --tw-brightness: ;
-  --tw-contrast: ;
-  --tw-grayscale: ;
-  --tw-hue-rotate: ;
-  --tw-invert: ;
-  --tw-saturate: ;
-  --tw-sepia: ;
-  --tw-drop-shadow: ;
-  --tw-backdrop-blur: ;
-  --tw-backdrop-brightness: ;
-  --tw-backdrop-contrast: ;
-  --tw-backdrop-grayscale: ;
-  --tw-backdrop-hue-rotate: ;
-  --tw-backdrop-invert: ;
-  --tw-backdrop-opacity: ;
-  --tw-backdrop-saturate: ;
-  --tw-backdrop-sepia: ;
 }
```

**Reply by fdarkaou** [reply]
Yes indeed... Thanks for pointing that out!

Any idea how I can avoid this?

**Reply by wongjn** [reply]
It could be due to some CSS minimization process within Nuxt.js. Consider looking at experimenting with any settings that could affect this.

**Reply by fdarkaou** [reply]
I have tried a bunch of things to avoid CSS minimization, purging of CSS and so on but nothing really changes. 

Somehow I have managed to get it working by accident on some of my pages. So currently you have some pages where the gradients work, and other ones where they don't and I have no clue at all why that is the case ..

**Reply by Digimonger** [reply]
I am facing the same issue. What can be done to fix it?
